pkg/authenticator/helmsecret.go

```diff
@@ -0,0 +1,33 @@
+package authenticator
+
+import (
+	"os"
+)
+
+type helmsecret struct {
+}
+
+var _ Authenticator = (*helmsecret)(nil)
+
+func NewHelmSecret() *helmsecret {
```

Comment:
Nit: Try to be consistent with your capitalization. That is, pick one of either helmSecret or helmsecret, but not both. I prefer helmSecret, because while I'm all for a good [compound word](https://en.wikipedia.org/wiki/English_compound), I don't think helmsecret qualifies. :)